src/main/java/io/netty/incubator/codec/http3/QpackDecoder.java

```diff
@@ -112,18 +386,93 @@ private CharSequence decodeStringLiteral(ByteBuf in, int length, boolean huffman
+        //            Error
+        //         ReqInsertCount -= FullRange
+        //
+        //      # Value of 0 must be encoded as 0.
+        //      if ReqInsertCount == 0:
+        //         Error
+        if (encodedInsertCount == 0) {
+            return 0;
+        }
+        if (encodedInsertCount > fullRange) {
+            throw INVALID_REQUIRED_INSERT_COUNT;
+        }
+
+        long maxValue = dynamicTable.insertCount() + maxEntries;
+        long maxWrapped = Math.floorDiv(maxValue, fullRange) * fullRange;
+        long requiredInsertCount = maxWrapped + encodedInsertCount - 1;
+
+        if (requiredInsertCount > maxValue) {
+            if (requiredInsertCount <= fullRange) {
+                throw INVALID_REQUIRED_INSERT_COUNT;
+            }
+            requiredInsertCount -= fullRange;
+        }
+        if (requiredInsertCount == 0) {
+            throw INVALID_REQUIRED_INSERT_COUNT;
+        }
+        return toIntOrThrow(requiredInsertCount);
+    }
+
+    private int decodeBase(ByteBuf buf, int requiredInsertCount) throws QpackException {
+        // https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#section-4.5.1
+        //   0   1   2   3   4   5   6   7
+        // +---+---------------------------+
+        // | S |      Delta Base (7+)      |
+        // +---+---------------------------+
+        final boolean s = (buf.getByte(buf.readerIndex()) & 0b1000_0000) == 0b1000_0000;
+        final int deltaBase = decodePrefixedIntegerAsInt(buf, 7);
+        assert deltaBase >= 0;
+        // https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-base
+        return s ? requiredInsertCount - deltaBase - 1 : requiredInsertCount + deltaBase;
+    }
+
+    private boolean shouldWaitForDynamicTableUpdates(int requiredInsertCount) throws QpackException {
+        if (requiredInsertCount > dynamicTable.insertCount()) {
+            if (blockedStreamsCount == maxBlockedStreams - 1) {
+                throw MAX_BLOCKED_STREAMS_EXCEEDED;
+            }
+            return true;
         }
-        throw NAME_ILLEGAL_INDEX_VALUE;
+        return false;
     }
 
-    private static QpackHeaderField getIndexedHeader(int index) throws QpackException {
-        if (index < QpackStaticTable.length) {
-            return QpackStaticTable.getField(index);
+    private void sendInsertCountIncrementIfRequired(QuicStreamChannel qpackDecoderStream) {
+        final int insertCount = dynamicTable.insertCount();
+        final List<Runnable> blockedStreams = this.blockedStreams.get(insertCount);
+        if (blockedStreams != null) {
+            for (Runnable blockedStream : blockedStreams) {
+                blockedStream.run();
+            }
+        }
+        if (stateSyncStrategy.entryAdded(insertCount)) {
+            // https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-insert-count-increment
+            final ByteBuf incr = qpackDecoderStream.alloc().buffer();
```

Comment:
I think we can do better here and just allocate a buffer with capacity of 8 as this is the "worst-case" space needed